system/src/system_sleep.cpp

```diff
@@ -47,19 +47,29 @@ using namespace particle;
 
 static bool system_sleep_network_suspend(network_interface_index index) {
     bool resume = false;
-    // Disconnect from network
-    if (network_connecting(index, 0, NULL) || network_ready(index, 0, NULL)) {
-        if (network_connecting(index, 0, NULL)) {
-            network_connect_cancel(index, 1, 0, 0);
+
+// FIXME: Dirty hack. For platforms support both Wi-Fi and Cellular interfaces,
+// we simply bypass the Wi-Fi interface 
+#if HAL_PLATFORM_WIFI && HAL_PLATFORM_CELLULAR
```

Comment:
Can you comment on why we need this? `network_connecting(index, 0, NULL) || network_ready(index, 0, NULL)` will be `false` on e.g. Tracker.

Reply:
It actually return `true` if the Cellular is connected to the tower, i.e. IP is obtained in system network manager and is specified as a wakeup source, in which case `network_ready()` returns `true`.

Reply:
I think that's not it. The problem is in how `network_connecting` is implemented on Gen 3 platforms. We will return `true` on a Tracker for WiFi interface if Cellular is connected or connecting or any other interface e.g. Ethernet.

Reply:
I think we can add a simple method to network manager to check interface flags:

`IFF_UP && !IFF_LOWER_UP` - would indicate `connecting` state.

Reply:
Yes, you're right. Also I just find an inconsistent state before/after sleep, which is when the application just turns on the modem to send AT commands manually and then make device enter sleep mode, after waking up the modem won't be turned on, since the `resume` flag isn't set before entering sleep mode. Considering fix it in the PR as well?

Reply:
Yes, let's add the fix in this PR.

Reply:
Fixed.